### PR TITLE
fix: redis caching

### PR DESCRIPTION
**Pull request by dejwi**
### Task Description
[SCOUT-290](https://playmakerpro.atlassian.net/browse/SCOUT-290?atlOrigin=eyJpIjoiZGE3MDhjNWRkYmM1NDFjZDlhN2IxNzFhNWM0NDUxZTUiLCJwIjoiaiJ9)

- fix user subscription caching

### CODE REVIEW

File: src/modules/user-subscriptions/user-subscriptions.service.ts [hunk 1]
```diff
@@ -27,10 +27,20 @@ export class UserSubscriptionsService {
     @InjectRedis() private readonly redis: Redis,
   ) {}
 
+  private getFormattedRedisKey(userId: string) {
+    return `user:${userId}:subscriptions`;
+  }
+
+  private async deleteFormattedFromCache(userId: string) {
+    await this.redis.del(this.getFormattedRedisKey(userId));
```
Comment (jakubfiglak) [inline src/modules/user-subscriptions/user-subscriptions.service.ts:35]:
```suggestion
    return this.redis.del(this.getFormattedRedisKey(userId));
```

File: src/modules/user-subscriptions/user-subscriptions.service.ts [hunk 2]
```diff
@@ -210,6 +223,11 @@ export class UserSubscriptionsService {
     await this.prisma.competitionGroupsOnUserSubscriptions.deleteMany({
       where: { subscriptionId: id },
     });
-    return this.prisma.userSubscription.delete({ where: { id }, include });
+    const deletedSubscription = await this.prisma.userSubscription.delete({
+      where: { id },
+      include,
+    });
+    this.deleteFormattedFromCache(deletedSubscription.userId);
```
Comment (jakubfiglak) [inline src/modules/user-subscriptions/user-subscriptions.service.ts:230]:
```suggestion
    await this.deleteFormattedFromCache(deletedSubscription.userId);
```

**Review by jakubfiglak — APPROVED** [review]
Good work, little tweaks suggested

File: src/modules/user-subscriptions/user-subscriptions.service.ts (hunk 1 above)
Comment (dejwi) [reply src/modules/user-subscriptions/user-subscriptions.service.ts:35]: check my reply at #135

**Comment by jakubfiglak** [comment]
This PR is approved, feel free to merge it

**Pull request merged**